### ISSUE TITLE
Retune LightGBM hyperparameters for all GB models after full-set predictor changes

### ISSUE THREAD

**Issue by frankbuckley** [dseinternational/language-reading-predictors#169]
> [!NOTE]
> Drafted by a LLM-based AI tool (Codex/GPT-5).

## Summary

Retune the LightGBM hyperparameters for all gradient-boosting models after recent changes to the GB modelling workflow, especially the retirement of hard feature selection in favour of full-set predictor ranking.

Many GB model modules still describe their current parameters as retained from earlier pruned-set Optuna tuning and marked `retune-pending`. Those settings are no longer clearly matched to the current full predictor sets, so we should run a reviewed retune and promote the resulting parameters into the declarative model definitions.

## Scope

Retune all registered GB models:

- Gain models: `lrpgbg01` through `lrpgbg22` (22 models)
- Level models: `lrpgbl01` through `lrpgbl28` (28 models)

Total: 50 models.

Current code paths of interest:

- `scripts/tune_model.py` — single-model Optuna tuning
- `scripts/fit_model.py` — GB fitting and reporting
- `src/language_reading_predictors/models/lrpgb*.py` — model definitions and committed params
- `tests/test_borrowed_params.py` — borrowed-parameter guard that will need updating if target-specific retuning retires borrowing
- `docs/models/README.md` and per-model report templates — prose may need updating from `retune-pending` / borrowed params to tuned full-set params

## Proposed Tuning Policy

Use the existing tuner unless the pilot exposes a problem:

```bash
python scripts/tune_model.py {model_id} --n-trials 150 --scoring mae --lgbm-objective mae
```

Rationale:

- The GB layer is mainly for robust predictor ranking, not for claiming causal or best possible predictive performance.
- Existing model prose generally frames the committed params as MAE-tuned.
- MAE is less dominated by a few high-scoring or high-gain children than RMSE.
- The tuner already uses `GroupKFold` by `subject_id` and an inner grouped early-stopping split, which matches the leakage constraints of the longitudinal data.

Open decision: decide whether any prediction-focused model should also receive an RMSE-tuned companion. If so, keep that separate from the ranking-oriented primary GB models.

### 1. Pilot the retune

Run a small pilot before launching all 50 models:

- [ ] `lrpgbg12` — word-reading gain
- [ ] `lrpgbl12` — word-reading level
- [ ] `lrpgbg06` — expressive-vocabulary gain
- [ ] `lrpgbl06` — expressive-vocabulary level

For each pilot model, inspect:

- `output/tuning/{model_id}/best_params.json`
- `output/tuning/{model_id}/trials.csv`
- `output/tuning/{model_id}/study_summary.json`

Check for boundary solutions, suspiciously small or large `n_estimators`, high CV spread, or worse performance than the current committed parameters.

### 2. Add batch orchestration if needed

`scripts/tune_model.py` currently tunes one model at a time. Add a small batch runner or extend the script so the full retune can be resumed safely.

Suggested requirements:

- [ ] tune all GB models or selected families (`gain`, `level`, `core`, `exploratory`)
- [ ] skip completed models unless `--force` is passed
- [ ] write a run manifest with model id, command args, git commit hash, start/end time, status, and output path
- [ ] continue after failures and report them at the end
- [ ] support a dry-run mode that lists planned models and commands

### 3. Run full tuning in waves

Suggested order:

- [ ] Core GB outcomes `01`–`16`
- [ ] Exploratory speech, memory, articulation, and language-sample models `17`–`28`
- [ ] Any failed or unstable models after review

### 4. Review tuning outputs

Before promoting params, create a review table for each model with:

- current committed params summary
- new Optuna best params
- old vs new dev/reporting CV metrics where available
- best trial score and CV standard deviation
- `n_estimators` / mean best iteration
- note on whether the chosen params are accepted, rejected, or need rerun

Important: do not treat tiny metric improvements as decisive. With small grouped samples, Optuna can optimise noise. Prefer stable, interpretable, and non-pathological parameter sets.

### 5. Promote accepted params into model modules

For each accepted model:

- [ ] update the `_LGBM_MAE_PARAMS` dict in the corresponding model module
- [ ] update comments and `notes` strings that say `retune-pending`
- [ ] remove or revise borrowed-parameter wording
- [ ] update `tests/test_borrowed_params.py` if target-specific tuning deliberately breaks current borrowing groups

Known borrowed groups currently guarded by tests:

```text
lrpgbg02 -> lrpgbg01, lrpgbg03, lrpgbg04
lrpgbg09 -> lrpgbg11
lrpgbl02 -> lrpgbl01, lrpgbl03, lrpgbl04
lrpgbl09 -> lrpgbl11
```

These should either be retired by target-specific tuning or explicitly kept with a documented reason.

### 6. Fit and validate after promotion

Run at least:

```bash
python scripts/fit_model.py all --config dev
pytest tests/test_models.py tests/test_borrowed_params.py
```

If compute allows before merge, also run:

```bash
python scripts/fit_model.py all --config reporting
```

Review the all-model summary for regressions or failed fits.

### 7. Document the retune

Add a dated note under `notes/` recording:

- tuning policy and command template
- model list and run manifest location
- failures, reruns, and exceptions
- which borrowed params were retired or retained
- summary of old vs new metrics
- any models whose params were not promoted and why

If the note is AI-drafted or substantially AI-edited, include the required project AI-authorship label at the top.

## Acceptance Criteria

- [ ] All 50 GB models have been considered for retuning.
- [ ] Accepted tuned parameters are committed in the relevant `lrpgb*.py` modules.
- [ ] `retune-pending` prose is removed or replaced where no longer true.
- [ ] Borrowed-parameter tests and annotations are reconciled.
- [ ] A dated retune note documents policy, commands, outputs, exceptions, and before/after summary.
- [ ] Dev fit for all GB models completes successfully.
- [ ] Required tests pass:

```bash
pytest tests/test_models.py tests/test_borrowed_params.py
ruff check src/
npm run format:check
npm run spellcheck
```

## Risks and Checks

- **Overfitting the tuning process:** Small grouped samples mean Optuna can chase fold noise. Require review before promotion.
- **Metric mismatch:** MAE is appropriate for robust ranking, but RMSE may be relevant for explicitly prediction-focused variants. Keep those purposes separate.
- **Silent parameter drift:** Updating only source models while leaving borrowed models unchanged can create misleading duplicate params. Reconcile `tests/test_borrowed_params.py` deliberately.
- **Compute cost:** A full 50-model, 150-trial run may be substantial. Use pilot and resumable batch orchestration before launching the full run.
